arch/risc-v/src/mpfs/mpfs_opensbi.c

```diff
@@ -51,6 +55,15 @@
 #define MPFS_PMP_DEFAULT_ADDR      0xfffffffff
 #define MPFS_PMP_DEFAULT_PERM      0x000000009f
 
+#define MPFS_DOMAIN_MAX_COUNT                4
+#define MPFS_DOMAIN_REGION_MAX_COUNT        12
+
+#define MPFS_SEL4_HART                1
```

Comment:
From these; we could generalize this to be up-streamable. That is, removing all references to LINUX & SEL4, and replace these by defines.

- mpfs_domain_init function could be moved off from this file into "board_domain_init" defined under board specific ones (e.g. boards/ssrc/saluki-v1/src/mpfs_domains_init.c )
- DOMAIN_INIT_ENABLED could be just a board-specific flag CONFIG_BOARD_OPENSBI_DOMAINS
- that is, in this file, only addition would be 
#ifdef CONFIG_BOARD_OPENSBI_DOMAINS
  .domains_init   = board_domains_init,
#endif